.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v2.6.2"
```

Comment:
Yeah, I'm used to seeing that warning in other projects, it's annoying.

I'm not familiar with pre-commit autoupdate. How does that work?

I was going based on the docs (the above was a copy + paste). But I can't find which docs now 😅 Either they've changed or I'm looking [in the wrong places](https://prettier.io/docs/en/precommit.html#option-3-pre-commithttpsgithubcompre-commitpre-commit)..

Reply:
`pre-commit autoupdate` fills in the newest version info into the `.pre-commit-hooks` file (and probably updates the installed pre-commit hooks). So, as soon as a commit hook is out of date a pre-commit hook user should be alerted to that and it will be easy enough to update.

I saw that, too on the prettier docs and then, because pre-commit complained, I changed it.

Reply:
So to install the latest tools used in the pre-commit hooks, one would have to run `pre-commit autoupdate` before installing the hooks. I could add that to the documentation and ask to alert this repo if someone tool requires a version update?

Reply:
after some googling, I created one. But there is a caveat - should this really run for every commit? I don't think so. Why? This pre-commit hook requires internet access. Normally, I like to restrict pre-commit hooks to checks that run locally. Also, this check will make a commit fail in every branch that is not pre-commit specific. So if someone implements a feature the pre-commit hooks might fail because a dependency had a new release. This will "force" a developer to convolute the pre-commit hook update into the current feature branch or manually fiddle with the pre-commit hook installation. 

But here you go, if you want to try that:

`.pre-commit-hooks`
```
repos:
  - repo: local
    hooks:
      - id: pre-commit-autoupdate
        name: pre-commit autoupdate
        description: This checks for updates for pre-commit hooks.
        entry: pre-commit autoupdate
        language: system
        pass_filenames: false
  - repo: https://github.com/psf/black
    rev: 22.3.0
    hooks:
      - id: black
  - repo: https://github.com/pycqa/isort
    rev: 5.10.1
    hooks:
      - id: isort
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: "v2.6.2"
    hooks:
      - id: prettier
```

Reply:
Awesome! 👏🏻 

Hmm, ok then. How about having this run on GitHub Actions CI only?

Reply:
I think if it's not blocking then it won't be noticed (no one ever looks at the output from a ✅).

Reply:
I would wait with that and see if https://github.com/orgs/github-community/discussions/15452 goes somewhere. It's specifically about how to show and handle GA failure in the PR UI. Additional community interaction would be welcome, too ;-)

Reply:
Interesting! Yeah let's skip over this for now. On reflection I think it'll be caught anyway as the CI is running separately to precommit (for now!) and that will be using the latest versions of things. So it amounts to the same thing.